content/00.build/30.zksync-sso/25.interfaces.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Interfaces
+description: Get familiar with ZKsync SSO interfaces
+---
+
+This document will explicitly list each interface method supported by the super-set account and module, along with brief descriptions of each method.
+It is important to clarify the distinction between the module interface and the account interface:
+the module interface specifies the methods that external contracts need to implement,
+whereas the account interface defines the methods that the smart account must support.
+
+## ZKsync SSO specific functions
+
+- `validateTransaction` - accepts a signature and the full transaction to execute, including paymaster information.
+- `executeTransaction` - invoked by ZKsync's entry point to run a verified transaction
+- `payForTransaction` and `prepareForPaymaster` - paymaster parsing functions called by the entry point to handle transaction fees and paymaster data.
+
+## Account functions from 7579
+
+- `executeFromExecutor` - a special entry point for modules that have been granted permission to execute within the smart account.
+- `accountId` - returns a unique identifier for the contract type
+- `supportsExecutionMode` - checks if a specific type of call data format is supported via entry points
+- `supportModule` - verifies if a module type (1-4) is available on the account
+- `installModule` - adds a module to the account
+- `uninstallModule` - removes a module from the account
+- `isModuleInstalled` - checks if a module is installed on the account
+
+## Hooks
+
+- `preCheck` - called before the execution flow begins
+- `postCheck` - called after the execution flow completes
+
+## Sessions
+
+Each allowed `(address, selector)` pair will have an assigned `FunctionPolicy` with all the applied constraints.
+`FunctionPolicy` will have a list of constraints on its arguments, as well as optional cumulative limits and allowances.
```

Comment:
This section is taken from an already quite outdated design doc, and should be updated. Should also include descriptions of all structs, their fields and methods that session key validator exposes.

Reply:
I updated this to just include the typescript interfaces and enums, which is better for external developers. Including our internal module interfaces isn't important as I totally misunderstood how different the context was for this vs our past interface docs.